From 236e0e76992ef7da7b2a7139655f77c31f376100 Mon Sep 17 00:00:00 2001
From: Steven Siloti <ssiloti@gmail.com>
Date: Wed, 11 Apr 2018 20:18:01 -0700
Subject: [PATCH] pass default constructed boost::function instead of NULL

Clang 6 defines NULL as nullptr which breaks implicit construction
of boost::function from NULL.
---
 src/kademlia/dht_tracker.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/kademlia/dht_tracker.cpp b/src/kademlia/dht_tracker.cpp
index 1831fbe6fc..4048e47ffe 100644
--- a/src/kademlia/dht_tracker.cpp
+++ b/src/kademlia/dht_tracker.cpp
@@ -224,7 +224,10 @@ namespace libtorrent { namespace dht
 	void dht_tracker::get_peers(sha1_hash const& ih
 		, boost::function<void(std::vector<tcp::endpoint> const&)> f)
 	{
-		m_dht.get_peers(ih, f, NULL, false);
+		// TODO: Use `{}` instead of spelling out the whole function type when this is merged to master
+		m_dht.get_peers(ih, f
+			, boost::function<void(std::vector<std::pair<node_entry, std::string> > const&)>()
+			, false);
 	}
 
 	void dht_tracker::announce(sha1_hash const& ih, int listen_port, int flags
